### ISSUE TITLE
can't get it to work if chat buffer get deleted.

### ISSUE THREAD

**Issue by fkgruber** [chep/copilot-chat.el#40]
If for some reason the copilot chat gets deleted then it won't create a new one. Just complains:

No buffer named *Copilot-chat-prompt*

**Comment by Lyonsclay** [comment]
Are you using the command `copilot-chat-display`?

**Comment by chep** [comment]
> Are you using the command `copilot-chat-display`?

This may be a workaround, but I think there is a bug when you kill prompt buffer and then call functions like explain, custom prompt because they use the prompt buffer. I should check buffer existence.

**Comment by chep** [comment]
In fact it works here @fkgruber can you give me a recipe to reproduce the issue ?

**Comment by fkgruber** [comment]
Well when I first run copilot-chat-prompt it works. but then if I erase *Copilot-chat-prompt* and run copilot-chat-prompt it gives that error.

But yes, after running copilot-chat-display it creates that buffer again and everything works again.

**Comment by chep** [comment]
`copilot-chat-prompt` is not a function, I tried `copilot-chat-custom-prompt-selection` and `copilot-chat-explain` and it works here.

**Comment by fkgruber** [comment]
strange. Perhaps I'm using an old version?


copilot-chat-prompt is an interactive byte-compiled Lisp function in
‘copilot-chat.el’.

(copilot-chat-prompt)

Open Copilot Chat Prompt buffer.

[back]

**Comment by fkgruber** [comment]
(defun copilot-chat-prompt ()
  "Open Copilot Chat Prompt buffer."
  (interactive)
  (let ((buffer copilot-chat-prompt-buffer))
    (with-current-buffer buffer
      (copilot-chat-prompt-mode))
    (switch-to-buffer buffer)))

**Comment by chep** [comment]
Yes I think you are using an old version, it has been removed in this commit : https://github.com/chep/copilot-chat.el/commit/1a3c0d34bace88e9996c0e34db6d18de48dd0658#diff-9a9bb71b8ad6778e4c23a482cb5001fc275c7046f74a675ad4e6ff95b2e33aed
(not so old, 2 weeks ago)
Now, `copilot-chat-display` is the only function to open prompt and it prepares buffers correctly even if they have been destroyed.
